spec/support/webmock/court_data_adaptor/webmock.rb

```diff
@@ -192,4 +192,251 @@
+    ).to_return(
+      status: 422,
+      headers: { 'Content-Type' => 'application/json' },
+      body: { 'errors' => { 'maat_reference' =>
+                             ['1234567 has no common platform data created against Maat application.'] } }
+              .to_json
+    )
+  end
+
+  config.before(:each, :stub_v2_link_server_failure) do
+    stub_request(
+      :post, %r{/v2/laa_references}
+    ).to_return(
+      status: 500,
+      body: ''
+    )
+  end
+
+  config.before(:each, :stub_v2_link_cda_failure) do
+    stub_request(
+      :post, %r{/v2/laa_references}
+    ).to_return(
+      status: 424,
+      body: ''
+    )
+  end
+
+  config.before(:each, :stub_defendants_case_search) do
+    stub_request(
+      :get, %r{http.*/v2/prosecution_cases\?filter.*}
+    ).to_return(
+      status: 200,
+      headers: { 'Content-Type' => 'application/json' },
+      body: load_json_stub('cda/internal_v2_prosecution_cases_response.json')
+    )
+  end
+
+  config.before(:each, :stub_defendants_ref_search) do
+    stub_request(
+      :get, %r{http.*/v2/prosecution_cases\?filter.*}
+    ).to_return(
+      status: 200,
+      headers: { 'Content-Type' => 'application/json' },
+      body: load_json_stub('cda/internal_v2_prosecution_cases_response.json')
+    )
+  end
+
+  config.before(:each, :stub_defendants_name_search) do
+    stub_request(
+      :get, %r{http.*/v2/prosecution_cases\?filter.*}
+    ).to_return(
+      status: 200,
+      headers: { 'Content-Type' => 'application/json' },
+      body: load_json_stub('cda/internal_v2_prosecution_cases_response.json')
+    )
+  end
+
+  config.before(:each, :stub_no_v2_results) do
+    stub_request(:get, %r{http.*/v2/defendants})
+      .to_return(
+        status: 200,
+        body: '{ "defendant_summaries": [] }',
+        headers: { 'Content-Type' => 'application/json' }
+      )
+  end
+
+  config.before(:each, :stub_no_v2_case_results) do
+    stub_request(
+      :get, %r{http.*/v2/prosecution_cases\?filter.*}
+    ).to_return(
+      status: 200,
+      body: '{ "total_results": 0, "results": [] }',
+      headers: { 'Content-Type' => 'application/json' }
+    )
+  end
+
+  config.before(:each, :stub_defendants_failed_search) do
+    stub_request(:get, %r{http.*/v2/defendants})
+      .to_return(
+        status: 400,
+        body: '',
+        headers: { 'Content-Type' => 'application/json' }
+      )
+  end
+
+  config.before(:each, :stub_defendants_failed_case_search) do
+    stub_request(
+      :get, %r{http.*/v2/prosecution_cases\?filter.*}
+    ).to_return(
+      status: 400,
+      headers: { 'Content-Type' => 'application/json' },
+      body: ''
+    )
+  end
+
+  config.before(:each, :stub_defendants_cda_failed) do
+    stub_request(:get, %r{http.*/v2/prosecution_cases\?filter.*})
+      .to_return(
+        status: 500,
+        body: '',
+        headers: { 'Content-Type' => 'application/json' }
+      )
+  end
+
+  config.before(:each, :stub_unlink_v2) do
```

Comment:
this says stub_unlink_v2 but it refers to a API v1

Reply:
To be fair, the version is pretty redundant now that our code doesn't support the v1 unlinking API anyway, so I've removed it.